src/pages/resultsView/plots/PlotsTabSelectionHistory.ts

```diff
@@ -0,0 +1,93 @@
+import { AxisMenuSelection, PlotsTabGeneOption, PlotsTabOption } from "./PlotsTab";
+import autobind from "autobind-decorator";
+
+type Selection = {
+    gene?: PlotsTabGeneOption | undefined,
+    geneSet?: PlotsTabOption | undefined,
+    source?: PlotsTabOption | undefined,
+    treatment?: PlotsTabOption | undefined,
+
+}
+
+type FieldUpdater = (option: any) => void;
+
+export default class PlotsTabSelectionHistory {
+    private horizontal: Map<string, Selection> = new Map();
+    private vertical: Map<string, Selection> = new Map();
+
+    @autobind
+    public updateHorizontalFromSelection(newSelection: AxisMenuSelection): void {
+        this.updateAxisWithSelection(this.horizontal, newSelection);
+    }
+    
+    @autobind
+    public updateVerticalFromSelection(newSelection: AxisMenuSelection): void {
+        this.updateAxisWithSelection(this.vertical, newSelection);
+    }
+
+    private updateAxisWithSelection(axis: Map<string, Selection>, newSelection: AxisMenuSelection): void {
+        if (newSelection.dataType !== undefined) {
+            let selectionToUpdate = axis.get(newSelection.dataType)
+
+            if (selectionToUpdate === undefined) {
+                selectionToUpdate = {};
+            }
+
+            axis.set(newSelection.dataType, this.updateSelection(selectionToUpdate, newSelection))
+        }
+    }
+
+    private updateSelection(selectionToUpdate: Selection, newSelection: AxisMenuSelection): Selection {
+        selectionToUpdate.gene = newSelection.selectedGeneOption;
+        selectionToUpdate.geneSet = newSelection.selectedGenesetOption;
+        selectionToUpdate.source = newSelection.selectedDataSourceOption;
+        selectionToUpdate.treatment = newSelection.selectedTreatmentOption;
+        return selectionToUpdate;
+    }
+
+    @autobind
+    public updateHorizontalExternalSelection(
+        type: string,
+        gene: FieldUpdater,
+        geneSet: FieldUpdater,
+        source: FieldUpdater,
+        treatment: FieldUpdater,
+    ): void {
+        this.runSelectionUpdaters(this.horizontal, type, gene, geneSet, source, treatment);
+    }
+
+    @autobind
+    public updateVerticalExternalSelection(
+        type: string,
+        gene: FieldUpdater,
+        geneSet: FieldUpdater,
+        source: FieldUpdater,
+        treatment: FieldUpdater,
+    ): void {
+        this.runSelectionUpdaters(this.vertical, type, gene, geneSet, source, treatment);
+    }
+
+    private runSelectionUpdaters(
+        axis: Map<string, Selection>,
+        type: string,
+        gene: FieldUpdater,
+        geneSet: FieldUpdater,
+        source: FieldUpdater,
+        treatment: FieldUpdater,
+    ) {
+        if (!axis.get(type)) {
+            return;
+        } 
+
+        const tuples = [
```

Comment:
can you rename `tuples` to something that describes what the data is, rather than the data type?